### ISSUE TITLE
Add an option of sending "encourage balloon"

### ISSUE THREAD

**Issue by bhscer** [DOMjudge/domjudge#2778]
### Description of the enhancement request

In ACM contests, it's common to stop distributing balloons during the scoreboard freeze. However, to create a more supportive and motivating environment, I propose adding "encouragement balloons" for teams with fewer solved problems. These balloons would offer a morale boost and enhance the overall competition experience, particularly for teams struggling to advance.

### The goal you want to achieve

During the freeze period, teams that have solved fewer problems than a set threshold (x) can still receive balloons for any new problems they solve, until they exceed the threshold.

### Expected behaviour

1. Step 1
Add options in the contest settings to enable "encouragement balloons" and to specify a threshold for the number of solved problems. Teams with fewer solved problems than this threshold will receive the encouragement balloons.
2. Step 2
In balloon list page, we add a special mark showing whether this balloon is an "encouragement balloon".

### Any other information that you want to share?

I apologize for my poor English. I hope the developers can understand what I am trying to express. Thank you for your understanding!

**Comment by vmcj** [comment]
You can already do this when using the icpctools balloon utility as that tool has this builtin.

Before considering this I think this would need more people requesting this as it is a bit controversial within the community. So I'll label it as a `low prio` for now until we get more people requesting this to prevent against feature creep.

**Comment by bhscer** [comment]
> You can already do this when using the icpctools balloon utility as that tool has this builtin.
> 
> Before considering this I think this would need more people requesting this as it is a bit controversial within the community. So I'll label it as a `low prio` for now until we get more people requesting this to prevent against feature creep.

Thank you for your quick response! Actually, I'm using the balloon API to read the balloons and import them into my system. I noticed the option "Show balloons post-freeze." Does this mean that it will collect balloons during the freeze period, allowing me to use the balloon API to read the balloons and decide whether to distribute them or not?

**Comment by bhscer** [comment]
Sorry that I accidentally closed this issue....

**Comment by cubercsl** [comment]
> > You can already do this when using the icpctools balloon utility as that tool has this builtin.
> > 
> > Before considering this I think this would need more people requesting this as it is a bit controversial within the community. So I'll label it as a `low prio` for now until we get more people requesting this to prevent against feature creep.
> 
> Thank you for your quick response! Actually, I'm using the balloon API to read the balloons and import them into my system. I noticed the option "Show balloons post-freeze." Does this mean that it will collect balloons during the freeze period, allowing me to use the balloon API to read the balloons and decide whether to distribute them or not?

This is how I configured in EC Final

**Comment by cubercsl** [comment]
Yes, this is how our balloon system is currently implemented. But I also think it will be nice if Domjudge internal balloon system have this feature.

**Comment by Dup4** [comment]
> Yes, this is how our balloon printer system is currently implemented. But I also think it will be nice if Domjudge internal balloon system have this feature.

yes yes

BestCuber!

**Comment by adityabelgaonkar05** [comment]
Can I work on this issue, or is it not necessary anymore?

**Comment by vmcj** [comment]
I think for such cases it is better to use the CDS with the balloon printer as it already has this option. Implementing this is not hard but keeping it correct over time is hard. So we would only accept PRs which have full code coverage for this feature (and even than we might consider it code creep as there was not much interest since November).
